src/main/java/com/pusher/client/channel/impl/PresenceChannelImpl.java

```diff
@@ -126,13 +125,14 @@ private void handleSubscriptionSuccessfulMessage(final String message) {
         final List<String> ids = presenceData.ids;
         final Map<String, Object> hash = presenceData.hash;
 
-        // build the collection of Users
-        for (final String id : ids) {
-            final String userData = hash.get(id) != null ? GSON.toJson(hash.get(id)) : null;
-            final User user = new User(id, userData);
-            idToUserMap.put(id, user);
+        if (ids != null && !ids.isEmpty()) {
```

Comment:
Is the `ids.isEmpty()` check necessary? If an interable data structure like this is empty, then the for loop will just iterate zero times, since there's nothing to iterate over. Unless I'm missing something, the only check necessary here (and thank you for adding this!) is the check that ids is not null.

Reply:
It's my understanding the `for` loop will create a `Iterator` object even if it's empty so this `isEmpty()` check offers a performance improvement (all be it a tiny one). If you feel strongly this makes the code less readable then I'm happy to change it.

Reply:
Not at all, I guess it can offer a small performance benefit! I'll merge this PR in and release by EOD.